code/obj/item/cigarette.dm

```diff
@@ -257,8 +257,10 @@
 			user.visible_message("<span class='alert'><B>[user]</B> blows smoke right into <B>[target]</B>'s face![message_append]</span>", group = "[user]_blow_smoke_at_[target]")
 
 			var/mob/living/carbon/human/human_target = target
-			if (human_target && rand(1,5) == 1)
-				SPAWN(0) target.emote("cough")
+			if (human_target && prob(20))
+				target.emote("cough")
```

Comment:
should check the `issmokeimmune()` macro here (which covers checks for gas masks, internals, anaerobic, etc.)

Reply:
im new to this so how would i go about doing that?

Reply:
`if (human_target && !issmokeimmune(human_target) && prob(20))`